src/App.tsx

```diff
@@ -8,7 +8,69 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import './App.scss';
+
 export const App: React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [status, setStatus] = useState('All');
+
+  const loadTodos = useCallback(async () => {
+    try {
+      const todos = await getTodos();
+
+      setVisibleTodos(todos);
+    } catch (error) {
+      throw new Error();
+    }
+  }, []);
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const handleEraseInput = useCallback(() => {
+    setQuery('');
+  }, []);
+
+  const handleCloseModal = useCallback(() => {
+    setSelectedTodo(null);
+  }, []);
+
+  const handleSelected = (todo: Todo | null) => {
+    setSelectedTodo(todo);
+  };
+
+  const handleQuery = useCallback((search: string) => {
```

Comment:
`useCallback` and `useMemo` are also functions with some logic, so we need to think twice before we start using them. Try to refactor your code and leave only important `useCallback`




`useCallback` we use to prevent rendering for memoized component because we don't create a new link on this function

 `useMemo` we use to prevent rendering for memoized component because we don't create a new link on array or object, and the second reason is to prevent some hard operation of array or object


memoized component - a component wrapped with `React.memo`